specs/SharedBuffer.md

```diff
@@ -0,0 +1,333 @@
+  /// Share a shared buffer object with script of the iframe in the WebView.
+  /// The script will receive a `SharedBufferReceived` event from chrome.webview.
+  /// The event arg for that event will have the following properties:
+  ///   `sharedBuffer`: an ArrayBuffer object with the backing content from the shared buffer.
+  ///   `data`: an object as the result of parsing `additionalDataAsJson` as JSON string.
+  ///           This property will be `undefined` if `additionalDataAsJson` is nullptr or empty string.
+  ///   `source`: with a value set as `chrome.webview` object.
+  /// If a string is provided as `additionalDataAsJson` but it is not a valid JSON string,
+  /// the API will fail with `E_INVALIDARG`.
+  /// If `isReadOnlyToScript` is true, the script will only have read access to the buffer.
+  /// If the script tries to modify the content in a read only buffer, it will cause an access
+  /// violation in WebView renderer process and crash the renderer process.
+  /// If the shared buffer is already closed, the API will fail with `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`.
+  /// 
+  /// The script code should call `chrome.webview.releaseBuffer` with
+  /// the shared buffer as the parameter to release underlying resources as soon
+  /// as it does not need access to the shared buffer any more.
+  ///
+  /// Sharing a buffer to script has security risk. You should only share buffer with trusted site.
+  /// If a buffer is shared to a untrusted site, possible sensitive information could be leaked.
+  /// If a buffer is shared as modifiable by the script and the script modifies it in an unexpected way,
+  /// it could result in corrupted data that might even crash the application. 
+  HRESULT PostSharedBufferToScript(
+    [in] ICoreWebView2SharedBuffer* sharedBuffer,
+    [in] BOOL isReadOnlyToScript,
+    [in] LPCWSTR additionalDataAsJson);
+}
+
+```
+
+## .NET
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    class CoreWebView2Environment
+    {
+        public CoreWebView2SharedBuffer CreateSharedBuffer(ulong size);
+    }
+
+    class CoreWebView2SharedBuffer : System.IDisposable
+    {
+        public ulong Size { get; };
+        
+        /// The raw memory address of the buffer.
+        /// You can cast it to pointer to real data types like byte* to access the memory
+        /// from `unsafe` code region.
+        /// Normal app should use `GetStream` to get a Stream object to access the buffer.
+        public IntPtr Buffer { get; };
+        
+        /// The native file mapping handle of the shared memory of the buffer.
+        /// Normal app should use `GetStream` to get a Stream object to access the buffer.
+        /// For advanced scenario, you could use native APIs to duplicate this handle to
+        /// another application process and create a mapping from the duplicated handle in
+        /// that process to access the buffer from that separate process.
+        public IntPtr Handle { get; };
```

Comment:
Maybe `DangerousGetHandle()` to align with SafeHandle?

Reply:
Will change to DangerousGetHandle(), if no objections. Should we also use this name for C++ API, or just GetHandle()?

Reply:
See above.